xarray/core/variable.py

```diff
@@ -1592,19 +1593,39 @@ def _unstack_once(
         else:
             dtype = self.dtype
 
-        data = np.full_like(
-            self.data,
-            fill_value=fill_value,
-            shape=new_shape,
-            dtype=dtype,
-        )
+        if sparse:
+            # TODO: how do we allow different sparse array types
```

Comment:
Do you know off-hand whether we need the separate code path now https://github.com/pydata/sparse/issues/422 is added? (If you don't, no need to spend time on it)

Reply:
I tried removing that but got stuck at some item assignment part that sparse is yet to support, see #5315.